spec/models/user_spec.rb

```diff
@@ -387,4 +387,30 @@
       end
     end
   end
+
+  describe '#archive!' do
+    context 'when archiving a user' do
+      subject(:user) { FactoryBot.create(:user) }
+
+      let(:nil_attributes) do
+        %w[avatar_thumb_url email birthday]
+      end
+
+      before { user.archive! && user.reload }
+
+      it { expect(user.archive!).to be true }
```

Comment:
You already called archive in the before, so I think this statement is duplicate, since you archive an already archived user